code/modules/events/crystal_event.dm

```diff
@@ -0,0 +1,439 @@
+GLOBAL_LIST_INIT(small_wave, list(/obj/structure/crystal_portal/small=4, /obj/structure/crystal_portal/medium=1)) //5
+GLOBAL_LIST_INIT(medium_wave, list(/obj/structure/crystal_portal/small=4, /obj/structure/crystal_portal/medium=3, /obj/structure/crystal_portal/big=1)) //8
+GLOBAL_LIST_INIT(big_wave, list(/obj/structure/crystal_portal/small=5, /obj/structure/crystal_portal/medium=3, /obj/structure/crystal_portal/big=2, /obj/structure/crystal_portal/huge=1)) //11
+GLOBAL_LIST_INIT(huge_wave, list(/obj/structure/crystal_portal/small=7, /obj/structure/crystal_portal/medium=5, /obj/structure/crystal_portal/big=3, /obj/structure/crystal_portal/huge=2)) //17
+GLOBAL_LIST_EMPTY(crystal_portals)
+/*
+This section is for the event controller
+*/
+/datum/round_event_control/crystal_invasion
+	name = "Crystal Invasion : Small"
+	typepath = /datum/round_event/crystal_invasion
+	weight = 4
+	min_players = 15
+	max_occurrences = 3
+	earliest_start = 25 MINUTES
```

Comment:
lmao you're funny

Reply:
Why what's wrong with those?

Reply:
this can happen THREE TIMES IN A SINGLE ROUND